src/com/massivecraft/vampire/ListenerMain.java

```diff
@@ -603,6 +604,61 @@ public void foodBlood(EntityDamageEvent event)
 	}
 	
 	// -------------------------------------------- //
+	// BLOOD FLASK
+	// -------------------------------------------- //
+	
+	@EventHandler(priority = EventPriority.NORMAL)
+	public void bloodFlaskConsume(PlayerItemConsumeEvent event)
+	{
+		// If the item is a potion ...
+		ItemStack item = event.getItem();
+		if ( ! item.getType().equals(Material.POTION)) return;
+		
+		// ... and is a blood flask ...
+		if ( ! BloodFlaskUtil.isBloodFlask(item)) return;
+		
+		// ... get the blood amount ...
+		double amount = BloodFlaskUtil.getBloodFlaskAmount(item);
+		
+		// ... and is the blood vampiric?  ...
+		boolean isVampiric = BloodFlaskUtil.getBloodFlaskVampiric(item);
+		
+		// ... get the player ...
+		UPlayer vme = UPlayer.get(event.getPlayer());

```

Comment:
I'd actually call this variable uplayer. This plugin is quite old, nowadays we'd call it a MPlayer instead. But since we (currently) stick to this naming convention, call it uplayer please.